src/types.rs

```diff
@@ -547,3 +564,18 @@ new_secret_type![
     #[derive(Clone)]
     ResourceOwnerPassword(String)
 ];
+new_secret_type![
+    ///
+    /// Device code returned by the device authorization endpoint and used to query the token endpoint.
+    ///
+    #[derive(Clone, Deserialize, Serialize)]
+    DeviceCode(String)
+];
+new_type![
+    ///
+    /// User code returned by the device authorization endpoint and used by the user to authorize at
+    /// the verification URI.
+    ///
+    #[derive(Deserialize, Serialize)]
```

Comment:
any reason to omit `Clone` here?

Reply:
Not a good one.. I'll double check why this got lost.

Reply:
```rust
error[E0119]: conflicting implementations of trait `std::clone::Clone` for type `types::UserCode`:
   --> src/types.rs:579:14
    |
78  |         #[derive(Clone, Debug, PartialEq)]
    |                  ----- first implementation here
...
579 |     #[derive(Clone, Deserialize, Serialize)]
    |              ^^^^^ conflicting implementation for `types::UserCode`
    |
```
Looks like other `new_type!` calls also don't derive Clone, e.g.
```rust
new_type![
    ///
    /// Code Challenge Method used for [PKCE]((https://tools.ietf.org/html/rfc7636)) protection
    /// via the `code_challenge_method` parameter.
    ///
    #[derive(Deserialize, Serialize, Eq, Hash)]
    PkceCodeChallengeMethod(String)
];
```
on line 397.